cpp/daal/include/services/env_detect.h

```diff
@@ -27,7 +27,7 @@
 
 #include "services/base.h"
 #include "services/daal_defines.h"
-#include "services/internal/execution_context.h"
+#include "services/error_handling.h"
```

Comment:
Good question - I faced several cases similar to this actually and its not immediately clear. But basically there is a whole web of inclusion dependencies and removal of includes would lead to build fails. This one specifically is included via `#include "services/internal/execution_context.h"` which contains `#include "services/internal/sycl/execution_context.h"` which contains `#include "services/internal/sycl/error_handling_sycl.h"` which contains `#include "services/internal/sycl/error_handling_sycl.h"`. Lack of this inclusion leads to: `/cpp/daal/src/externals/service_rng_mkl.h(338): error: namespace "daal::services" has no member "Status"` during build. Status is defined in error_handling.h. It's chaos but I think this is the simplest resolution.

Reply:
Ouch. Fair enough